test/main.cpp

```diff
@@ -24,6 +24,16 @@ SOFTWARE.
 
 #include "../src/polylineencoder.cpp"
 
+static bool operator==(const PolylineEncoder::Point& l, const PolylineEncoder::Point& r)
+{
+    return l.longitude() == r.longitude() && r.latitude() == r.latitude();
```

Comment:
Nice, but it may give wrong results, as you check two doubles for equality. We should use more sophisticated comparison here like:
````C++
std::fabs(l.longitude() - r.longitude()) * 1000000000000 <= std::min(std::fabs(l.longitude()), std::fabs(r.longitude()));
````
The same for the latitude.

Reply:
slightly different way of implementation, done